### ISSUE TITLE
Restarting REC causes a crack on PLAY

### ISSUE THREAD

**Issue by nis65** [Daenou/ansible-multiroom-audio#41]
According to [my discussion with hifiberry](https://support.hifiberry.com/hc/en-us/community/posts/4415021634577), it is plain normal that a loud crack is produced on the DAC output when

* there is (at least one) application already playing audio to the DAC
* you start rec from the ADC (which causes a reinit of the "clock circuit" which is used in common by the DAC and the ADC).

At least for hifiberry HATs, this is a hardware limitation. 

The only solution is **not** to start REC as long as a PLAY is running. I therefore propose the following feature:

Starting an acable that records from the soundcard (in our arch: `dsnoop`) should be done in the following way:
* before starting, all **other** acables and all snapclients that write to the soundcard (in our arch) `dmix` device, are stopped. 
* then the acable can start, including the REC from the soundcard `dsnoop`.
* only then, all temporarily stopped acables and snapclients are started again.

This still produces a short period of silence when starting rec. But this would be much better than the current crack that sounds so bad that I fear it hurts the loudspeakers if I have the volume too high.

**Comment by nis65** [comment]
The final confirmation from hifiberry that this is a WONTFIX for their hardware is here (see link in first entry).

A workaround is therefore mandatory:
* either accept that our acable reading from hifiberry dsnoop MUST be stable (which it actually will be, when #40 is fixed) and close this issue as WONTFIX.
* build a logic as described above. tbd: most leight-weight implementation:
  * how do the hifiberry dsnoop rec apps (could be more than one, but unlikely) know which are all hifiberry dmix play apps (quite likely to me more than one) that must be briefly shut down and started only after rec has been started? additional ansible vars or logic to find that out automatically?
  * additional script logic or all doable with systemd unit dependencies only (would love this, but sounds hard). Other implementations?

**Comment by Daenou** [comment]
I'm all for building a login and solving the issue cleanly.

**Comment by nis65** [comment]
Current solution stops simply all snapclients whenever a "non bluetooth" acable is started. This causes a short silence. While this is annoying, it is much less annoying than the crack. This is the reasoning:

* `snapclients` should always output to the hardware `dmix` sink (and not a `snd_aloop` device or similar as this would ruin the synced multiroom experience), so these must be suspended in almost any reasonable use case.
* My only use case for another acable outputting to the hardware `dmix` sink (that should be stopped in addition to the snapclients) is a loop from the analog input directly to the analog output. And this acable must not be stopped (the "suspend" logic should not try to suspend itself)!
* Our current ansible  architecture would also allow to configure e.g. an mpd outputting to hardware and in such a config, its output should also be suspended to avoid the crack. Such use cases are not handled right now and restarting an acable still might provoke a crack (as the script may miss a process currently sending audio to hw out).

But good enough for me.
